contract_sale_generation/models/contract_line.py

```diff
@@ -40,7 +40,10 @@ def _prepare_sale_line(self, order_id=False, sale_values=False):
             )
             order_line.order_id = sale
         # Get other order line values from product onchange
+        uom = order_line['product_uom']
         order_line.product_id_change()
+        # Do set again the product_uom - the product_id_change function does reset it before
+        order_line['product_uom'] = uom
```

Comment:
@wpichler  I think you can put it in sale_line_vals.update() instead. What do you think ?